File: 3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.java
class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {

        int total = 0;
        for (int a : apple) total += a;

        // Find max capacity
        int maxCap = 0;
        for (int c : capacity) {
            if (c > maxCap) maxCap = c;
        }

        // Count frequencies
        int[] freq = new int[maxCap + 1];
        for (int c : capacity) {
            freq[c]++;
        }

        int used = 0;
        int sum = 0;

        // Traverse from largest capacity
        for (int cap = maxCap; cap >= 0; cap--) {
            while (freq[cap]-- > 0) {
                sum += cap;
                used++;
                if (sum >= total) {
                    return used;
                }
            }
        }
        return used;
    }
}

File: 3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.py
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        capacity.sort(reverse=True)
        total=sum(apple)
        temp=0
        for i in range(len(capacity)):
            temp+=capacity[i]
            if temp>=total:
                return i+1
        return i+1

        